### PR TITLE
feat: Support for multiple document yaml files

### PR DESCRIPTION
**Pull request by sorenmat**
Motivation for this is that the deployment target has a kapeta.yml file with multiple documents in it
Also moved to typescript

### CODE REVIEW

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -7,9 +7,9 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
```
Comment (wejendorp) [inline .github/workflows/pr-check.yml:11]: Both checkout and setup-node are available in a v4

File: codegen.js [hunk 2]
```diff
@@ -1,33 +1,3 @@
-#!/usr/bin/env node
-/**
- * Copyright 2023 Kapeta Inc.
- * SPDX-License-Identifier: MIT
- */
+// Temp file because blockctl only regards index.js instead of package.main
```
Comment (wejendorp) [inline codegen.js:1]: Looks like we removed the execute bit. Is this on purpose?

Comment (wejendorp) [reply codegen.js:1]: And shouldn't this file be named `index.js` then?

File: package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "command": "codegen",
     "version": "0.0.40",
     "description": "Command for generating code from kapeta YML files",
-    "main": "index.js",
+    "main": "dist/index.js",
```
Comment (wejendorp) [inline package.json:6]: Is it `dist/index.js` or `dist/src/index`? 👀 Codegen above says `src` too

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,110 @@
+{
+  "compilerOptions": {
```
Comment (hofmeister) [inline tsconfig.json:2]: There is a standard node one we use for this

Comment (hofmeister) [reply tsconfig.json:2]: Cant remember to name exactly but google should know - something like node18

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env node
+import {DefinitionInfo} from "@kapeta/local-cluster-config";
+
+/**
+ * Copyright 2023 Kapeta Inc.
+ * SPDX-License-Identifier: MIT
+ */
+
+
+const KapetaCommand = require('@kapeta/kap-command');
```
Comment (hofmeister) [inline src/index.ts:10]: Import? Or that didnt work?

**Review by hofmeister — APPROVED** [review]
Minor things

File: src/actions/generate.test.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        };
+        const blocks = blockYamlFilter(yaml);
+        assert.equal(blocks.length, 2);
+        assert.ok(blocks[0].filename === 'test.yaml');
+        assert.ok(blocks[0].content.kind === 'Block');
+        assert.ok(blocks[1].filename === 'test.yaml');
+        assert.ok(blocks[1].content.kind === 'Block2');
+    });
+
+    it('should return an array containing a block for a YAML file that contains a block with a core prefix', () => {
```
Comment (wejendorp) [inline src/actions/generate.test.ts:51]:
```suggestion
    it('should filter out blocks with core prefix', () => {
```

File: src/actions/generate.ts [hunk 7]
```diff
@@ -12,24 +12,37 @@ const YAML = require('yaml');
 const ClusterConfiguration = require('@kapeta/local-cluster-config').default;
 
 const CORE_PREFIX = 'core/';
+// define a type for the block
+export interface BlockFile {
+    filename: string;
+    content: Document[];
```
Comment (wejendorp) [inline src/actions/generate.ts:18]: I think this is accidentally using the HTML Document type. Seems like a bug.

File: src/actions/generate.ts [hunk 8]
```diff
@@ -12,24 +12,37 @@ const YAML = require('yaml');
 const ClusterConfiguration = require('@kapeta/local-cluster-config').default;
 
 const CORE_PREFIX = 'core/';
+// define a type for the block
+export interface BlockFile {
```
Comment (wejendorp) [inline src/actions/generate.ts:16]: Shouldn't this be used instead of `any` in the generate signatures?

File: package.json (hunk 3 above)
Comment (sorenmat) [reply package.json:6]: I have a dist/index.js which is the src/index.ts file so I guess it should be fine

File: tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "@tsconfig/node18/tsconfig.json",
+  "compilerOptions": {
+    "target": "es2016",                               "module": "commonjs",                             "resolveJsonModule": true,                        "allowJs": true,                                  "outDir": "./dist",                               "esModuleInterop": true,                          "forceConsistentCasingInFileNames": true,         "strict": true,                                   "skipLibCheck": true                            },
```
Comment (wejendorp) [inline tsconfig.json:4]: Some might argue this could be made more readable.

**Review by wejendorp — APPROVED** [review]

**Pull request merged**